### ISSUE TITLE
Move to pytest from nose and add mypy

### ISSUE THREAD

**Issue by clintval** [r-bioinformatics/edgePy#19]
Consider wiring it all together with tox and tox-travis.

**Comment by clintval** [comment]
With #18 it seems we are accepting of optional type annotations.

If true, we should move our docstring format over to that codified in the Google Style Guide:

```python
def get_dataset_path(
    filename: Union[str, Path],
    dead_arg: Optional[Any]=None
) -> Path:
    """Example function with PEP 484 type annotations.

    Args:
        filename: The first parameter.
        dead_arg: The second parameter.

    Returns:
        The path to the dataset, may not really exist.

    Examples:
        >>> from module.io import get_dataset_path
        >>> str(get_dataset_path("GSE49712_HTSeq.txt.gz"))  # doctest:+ELLIPSIS
        '.../data/GSE49712_HTSeq.txt.gz'

    Notes:
        1. See ``module.rationalize`` for an equivalent method.

    """
    import module
    directory = Path(module.__file__).expanduser().resolve().parent
    return directory / 'data' / filename
```

- [ ] Ensure type hints are used (optional) and are correct
- [ ] Change `CONTRIBUTING.md` to incorporate these new standards
- [ ] Port to `pytest`
- [ ] Add a MyPy step to CI
- [ ] Change docstrings to Google Style Guide (removing redundant type annotation)

I can take this on, it will be quick whenever I do it.

**Comment by clintval** [comment]
This can be closed with merge of #18.
